Source/SwiftLintBuiltInRules/Rules/Idiomatic/FileNameRule.swift

```diff
@@ -12,6 +13,7 @@ struct FileNameRule: OptInRule, SourceKitFreeRule {
 
     func validate(file: SwiftLintFile) -> [StyleViolation] {
         guard let filePath = file.path,
+            !shouldExclude(filePath: filePath),
             case let fileName = filePath.bridge().lastPathComponent,
             !configuration.excluded.contains(fileName) else {
```

Comment:
Did you intentionally keep this check? I think it's a good way to ensure backwards compatibility even though an exclude name that is not a regular expression would match in `shouldExclude` already. We could make this part of `shouldExclude` then, probably the first check in there.